src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -317,26 +316,21 @@ def find_subscriptions_in_vm_with_msi(self, identity_id=None, allow_no_subscript
                 identity_type = MsiAccountTypes.user_assigned_resource_id
             else:
                 authenticated = False
+                from .azclierror import AzureResponseError
                 try:
                     msi_creds = MSIAuthenticationWrapper(resource=resource, client_id=identity_id)
                     identity_type = MsiAccountTypes.user_assigned_client_id
                     authenticated = True
-                except HTTPError as ex:
-                    if ex.response.reason == 'Bad Request' and ex.response.status == 400:
```

Comment:
`400 Bad Request` is not checked anymore?

Reply:
This remove is just a try for full test. I have added it back